src/software/backend/simulation/simulator_robot.h

```diff
@@ -0,0 +1,154 @@
+#pragma once
+
+#include <cinttypes>
+#include <optional>
+
+#include "software/backend/simulation/physics/physics_robot.h"
+extern "C"
+{
+#include "app/world/firmware_robot.h"
+}
+
+/**
+ * This class acts as a wrapper around a PhysicsRobot so that the PhysicsRobot
+ * can provide the interface of a FirmwareRobot.
+ *
+ * Because our firmware structs rely on C-style function pointers, we
+ * cannot use C++ constructs like std::function to provide to the structs.
+ * Therefore we use this class's static functions to provide the function
+ * pointers, since static functions work as C-style function pointers
+ * (because there is no instance associated with them).
+ *
+ * Whenever a caller needs to get a firmware struct or perform an operation, they
+ * need to set which robot they want to control so that all the static functions
+ * that have been provided to the firmware struct operate on the correct
+ * instantiated object. This is our workaround to maintain and simulate multiple
+ * "instances" of robot firmware at once.
+ */
+class SimulatorRobot
+{
+   public:
+    /**
+     * Sets the ID of the robot being controlled by this class
+     *
+     * @param id The ID of the robot to control
+     */
+    static void setRobotId(unsigned int id);
+
+    /**
+     * Sets the PhysicsRobots that can be controlled by this class
+     *
+     * @param robots the PhysicsRobots that can be controlled by this class
+     */
+    static void setPhysicsRobots(const std::vector<std::weak_ptr<PhysicsRobot>>& robots);
+
+    /**
+     * Creates a FirmwareRobot corresponding to the current PhysicsRobot
+     *
+     * @return a FirmwareRobot corresponding to the current PhysicsRobot
+     */
```

Comment:
I would _really_ prefer that we don't return raw pointers like this. Perhaps we could store the robots internally in a static table?

Reply:
Not sure what you mean about storing internally in a static table (don't see how that solves returning pointers).

I realized I can return a `unique_ptr` here instead since this isn't actual firmware, but gonna hold off on changes until I understand the table idea.

Reply:
Sorry, I think I misread some things here, returning a `unique_ptr` should be fine.

Reply:
Changed to return a `unique_ptr`. I had to define a `Deleter` struct, so please take a look at that.

Reply:
Oh very neat!